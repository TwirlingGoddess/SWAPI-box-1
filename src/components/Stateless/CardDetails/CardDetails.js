import React from 'react';
import { Link } from 'react-router-dom';
import '../Card/Card.scss';
import PropTypes from 'prop-types';

const CardDetails = (selectedCard) => {
  return (
    <div>
      <div className='selected card'>
        <Link to={`/${selectedCard.id}`} 
          className='favorite'>Selected Card</Link>
        <h1>{selectedCard.Name}</h1>
        <div className='pic'/>
      </div>
    </div>
  );
};

CardDetails.propTypes = {
  selectedCard: PropTypes.obj
};
export default CardDetails;