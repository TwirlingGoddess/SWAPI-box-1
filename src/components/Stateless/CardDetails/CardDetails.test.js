import React from 'react';
import { shallow, mount } from 'enzyme';
import CardDetails from '../CardDetails/CardDetails';

let mockProps = {
  data:{},
  id: 2,
  clicked: false,
  findCard: jest.fn()
};

describe('Card', () => {
  let cardDetails;
  
  beforeEach(() => {
   cardDetails = shallow(<CardDetails />, { disableLifecycleMethods: true });
  })

  it('matches snapshot', () => {
    expect(cardDetails).toMatchSnapshot();
  })

  it(" has a class of 'selectedCard' ", () => {
    const selectedStyle = cardDetails.find('.card');
    expect(selectedStyle.hasClass('selected')).toEqual(true);
  });
});
