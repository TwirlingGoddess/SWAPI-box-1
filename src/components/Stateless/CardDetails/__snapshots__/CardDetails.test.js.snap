// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="selected card"
>
        <Link
                className="favorite"
                replace={false}
                to="/undefined"
        >
                Selected Card
        </Link>
        <h1 />
        <div
                className="pic"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="favorite"
            replace={false}
            to="/undefined"
>
            Selected Card
</Link>,
          <h1 />,
          <div
            className="pic"
/>,
        ],
        "className": "selected card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Selected Card",
            "className": "favorite",
            "replace": false,
            "to": "/undefined",
          },
          "ref": null,
          "rendered": "Selected Card",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pic",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="selected card"
>
          <Link
                    className="favorite"
                    replace={false}
                    to="/undefined"
          >
                    Selected Card
          </Link>
          <h1 />
          <div
                    className="pic"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="favorite"
              replace={false}
              to="/undefined"
>
              Selected Card
</Link>,
            <h1 />,
            <div
              className="pic"
/>,
          ],
          "className": "selected card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Selected Card",
              "className": "favorite",
              "replace": false,
              "to": "/undefined",
            },
            "ref": null,
            "rendered": "Selected Card",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pic",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
